graphium/data/datamodule.py

```diff
@@ -1075,12 +1075,16 @@ def has_atoms_after_h_removal(smiles):
 
         """Filter data based on molecules which failed featurization. Create single task datasets as well."""
         self.single_task_datasets = {}
+
+        idx_none = []
         for task, args in task_dataset_args.items():
             # Find out which molecule failed featurization, and filter them out
-            idx_none = []
-            for idx, feat in enumerate(args["features"]):
-                if did_featurization_fail(feat):
+            for idx, (feat, labels, smiles) in enumerate(zip(args["features"], args["labels"], args["smiles"])):
+                if did_featurization_fail(feat) or found_size_mismatch(task, feat, labels, smiles):
                     idx_none.append(idx)
+        idx_none = list(set(idx_none))
+
+        for task, args in task_dataset_args.items():
```

Comment:
I don't think what you did makes sense. If I understand correctly, if the 4th molecule on task1 failed featurization, it will also remove the 4th molecule on task2. But these might be completely different molecules.

Reply:
You are right. After our discussion, it was not the filtering, but rather the unit test `tests/test_multitask_datamodule.py` that needed to be fixed here cde61e166572b94b8718f6dba5c3e39a03839503